### PR TITLE
Reports to warehouse

### PR DESCRIPTION
**Pull request by edasmalchi**
# Description

New table to consolidate and improve metric generation for reports. Replaces complex Siuba code in reports notebook and allows for visibility into service hour changes before the report generation process.

Resolves https://github.com/cal-itp/reports/issues/129

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation
- [ ] agencies.yml

## How has this been tested?

Ran for 2 months via dbt, that processed 1.4GB.

### CODE REVIEW

**Comment by edasmalchi** [comment]
will add yaml change and link reports repo pr before merging

**Review by lauriemerrell — CHANGES_REQUESTED** [review]
Eric to add YAML

File: warehouse/models/gtfs_views/_gtfs_views.yml
Comment (lauriemerrell) [inline warehouse/models/gtfs_views/_gtfs_views.yml:846]: I would refactor this a bit to be a generic description of the table content / purpose -- this description will persist in general, so table will no longer be "new".

File: warehouse/models/gtfs_views/reports_daily_service_routes_stops.sql
Comment (lauriemerrell) [inline warehouse/models/gtfs_views/reports_daily_service_routes_stops.sql:73]: stylistic optional thing: you could refactor the original CTE to just be the below instead of having dim_date separately. not a big deal but don't necessarily need to have such a minimal CTE as a separate item


```
dates AS (
    SELECT full_date AS service_date
    FROM {{ ref('dim_date') }}
),
```

File: warehouse/models/gtfs_views/reports_daily_service_routes_stops.sql
Comment (lauriemerrell) [inline warehouse/models/gtfs_views/reports_daily_service_routes_stops.sql:79]: question: do you actually need `use_for_report` here?

File: warehouse/models/gtfs_views/reports_daily_service_routes_stops.sql
Comment (lauriemerrell) [inline warehouse/models/gtfs_views/reports_daily_service_routes_stops.sql:63]: optional suggestion: this could probably be combined with the above (just add one more join)

**Review by lauriemerrell — CHANGES_REQUESTED** [review]
One specific change re: table description, otherwise looks ok to me... 

One stylistic thing is that this is kind of a lot of CTEs, if it were me I might try to combine a few just to make it a bit shorter, but I don't think it's a huge deal/requirement. [This post](https://discourse.getdbt.com/t/why-the-fishtown-sql-style-guide-uses-so-many-ctes/1091) says that they try to keep their queries to <100 lines of SQL, which might be a good ballpark goal.

File: warehouse/models/gtfs_views/reports_daily_service_routes_stops.sql
Comment (edasmalchi) [reply warehouse/models/gtfs_views/reports_daily_service_routes_stops.sql]: ah, probably not since the second join should handle it

File: warehouse/models/gtfs_views/reports_daily_service_routes_stops.sql
Comment (edasmalchi) [inline warehouse/models/gtfs_views/reports_daily_service_routes_stops.sql:63]: Is this the right way to combine those joins?

**Comment by edasmalchi** [comment]
thanks for the comments @lauriemerrell, tried to fix in the latest commit

**Review by atvaccaro — APPROVED** [review]
lgtm I think the distincts might be unnecessary tho

File: warehouse/models/gtfs_views/reports_daily_service_routes_stops.sql
Comment (lauriemerrell) [reply warehouse/models/gtfs_views/reports_daily_service_routes_stops.sql:63]: Yes, looks good to me!

**Review by lauriemerrell — APPROVED** [review]
LGTM! I see Andrew has at least one comment so I defer to him on that. Congrats on first dbt/SQL PR in data-infra! 🎉

**Comment by atvaccaro** [comment]
oh @edasmalchi have you set up pre-commit locally? it looks like they are failing in CI/CD

File: warehouse/models/gtfs_views/reports_daily_service_routes_stops.sql
Comment (edasmalchi) [reply warehouse/models/gtfs_views/reports_daily_service_routes_stops.sql]: good point, removed them here

**Pull request merged**
